crates/bevy_gizmos/src/primitives.rs

```diff
@@ -0,0 +1,1954 @@
+    gizmos: &'a mut Gizmos<'w, 's, T>,
+
+    direction: Direction2d, // direction the arrow points to
+
+    position: Vec2, // position of the start of the arrow
+    color: Color,   // color of the arrow
+}
+
+impl<T: GizmoConfigGroup> Direction2dBuilder<'_, '_, '_, T> {
+    /// set the position of the start of the arrow
+    pub fn position(mut self, position: Vec2) -> Self {
+        self.position = position;
+        self
+    }
+
+    /// set the color of the arrow
+    pub fn color(mut self, color: Color) -> Self {
+        self.color = color;
+        self
+    }
+}
+
+impl<'w, 's, T: GizmoConfigGroup> GizmoPrimitive2d<Direction2d> for Gizmos<'w, 's, T> {
+    type Output<'a> = Direction2dBuilder<'a, 'w, 's, T> where Self: 'a;
+
+    fn primitive_2d(&mut self, primitive: Direction2d) -> Self::Output<'_> {
+        Direction2dBuilder {
+            gizmos: self,
+            direction: primitive,
+            position: Default::default(),
+            color: Default::default(),
+        }
+    }
+}
+
+impl<T: GizmoConfigGroup> Drop for Direction2dBuilder<'_, '_, '_, T> {
+    fn drop(&mut self) {
+        let start = self.position;
+        let end = self.position + *self.direction;
+        self.gizmos.arrow_2d(start, end, self.color);
+    }
+}
+
+// circle 2d
+
+/// Builder for configuring the drawing options of [`Circle`].
+pub struct Circle2dBuilder<'a, 'w, 's, T: GizmoConfigGroup> {
+    gizmos: &'a mut Gizmos<'w, 's, T>,
+
+    radius: f32, // 2D circle to be rendered
+
+    center: Vec2, // position of the center of the circle
+    color: Color, // color of the circle
+}
+
+impl<T: GizmoConfigGroup> Circle2dBuilder<'_, '_, '_, T> {
+    /// Set the position of the center of the circle.
+    pub fn center(mut self, center: Vec2) -> Self {
+        self.center = center;
+        self
+    }
+
+    /// Set the color of the circle.
+    pub fn color(mut self, color: Color) -> Self {
+        self.color = color;
+        self
+    }
+}
+
+impl<'w, 's, T: GizmoConfigGroup> GizmoPrimitive2d<Circle> for Gizmos<'w, 's, T> {
+    type Output<'a> = Circle2dBuilder<'a, 'w, 's, T> where Self: 'a;
+
+    fn primitive_2d(&mut self, primitive: Circle) -> Self::Output<'_> {
+        Circle2dBuilder {
+            gizmos: self,
+            radius: primitive.radius,
+            center: Default::default(),
+            color: Default::default(),
+        }
+    }
+}
+
+impl<T: GizmoConfigGroup> Drop for Circle2dBuilder<'_, '_, '_, T> {
+    fn drop(&mut self) {
+        self.gizmos.circle_2d(self.center, self.radius, self.color);
+    }
+}
+
+// ellipse 2d
+
+/// Builder for configuring the drawing options of [`Ellipse`].
+pub struct Ellipse2dBuilder<'a, 'w, 's, T: GizmoConfigGroup> {
+    gizmos: &'a mut Gizmos<'w, 's, T>,
+
+    half_width: f32,  // Half-width of the ellipse
+    half_height: f32, // Half-height of the ellipse
+
+    center: Vec2, // Position of the center of the ellipse
+    color: Color, // Color of the ellipse
+}
```

Comment:
Ellipses should support rotation like other primitives. Imo every primitive (even the ones with a direction or normal) should be rotatable, as they don't have an inherent rotation associated with them